src/OpenTK.GLControl/GLControl.cs

```diff
@@ -191,32 +212,37 @@ protected override void OnHandleCreated(EventArgs e)
                 implementation.WindowInfo.Dispose();
             }
 
-            if (design_mode)
-            {
-                implementation = new DummyGLControl();
-            }
-            else
+            try
             {
-                implementation = new GLControlFactory().CreateGLControl(format, this);
-            }
+                if (design_mode)
+                    implementation = new DummyGLControl();
+                else
+                    implementation = new GLControlFactory().CreateGLControl(format, this);
 
-            context = implementation.CreateContext(major, minor, flags);
-            MakeCurrent();
+                context = implementation.CreateContext(major, minor, flags);
+                MakeCurrent();
 
-            if (!design_mode)
+                if (!design_mode)
+                    ((IGraphicsContextInternal)Context).LoadAll();
+
+                // Deferred setting of vsync mode. See VSync property for more information.
+                if (initial_vsync_value.HasValue)
+                {
+                    Context.SwapInterval = initial_vsync_value.Value ? 1 : 0;
+                    initial_vsync_value = null;
+                }
+            }
+            catch
```

Comment:
This should catch the specific exception type

Reply:
Agree. I will change this